example_cpp_smart_card_client_app/src/chrome_certificate_provider/types.h

```diff
@@ -76,20 +84,37 @@ enum class PinRequestErrorType {
 // Structure containing a certificate description.
 //
 // For the corresponding original JavaScript definition, refer to:
-// <https://developer.chrome.com/extensions/certificateProvider#type-CertificateInfo>.
-struct CertificateInfo {
+// <https://developer.chrome.com/extensions/certificateProvider#type-ClientCertificateInfo>.
+struct ClientCertificateInfo {
   std::vector<uint8_t> certificate;
-  std::vector<Hash> supported_hashes;
+  std::vector<Algorithm> supported_algorithms;
 };
 
-// Structure containing the data of the digest signing request.
+// Structure containing the parameters for the
+// chrome.certificateProvider.setCertificates() function.
 //
 // For the corresponding original JavaScript definition, refer to:
-// <https://developer.chrome.com/extensions/certificateProvider#type-SignRequest>.
-struct SignRequest {
+// <https://developer.chrome.com/extensions/certificateProvider#method-setCertificates>.
+struct SetCertificatesDetails {
+  google_smart_card::optional<int> certificates_request_id;
+  google_smart_card::optional<Error> error;
+  std::vector<ClientCertificateInfo> client_certificates;
+};
+
+// Structure containing the data of the signature request.
+//
+// Note that either |input| or |digest| will be set (but not both
+// simultaneously).
+//
+// For the corresponding original JavaScript definition, refer to:
+// <https://developer.chrome.com/extensions/certificateProvider#event-onSignatureRequested>
+// and
+// <https://developer.chrome.com/extensions/certificateProvider#event-onSignDigestRequested>.
+struct SignatureRequest {
   int sign_request_id;
+  std::vector<uint8_t> input;
   std::vector<uint8_t> digest;
```

Comment:
Should digest still be included here? It is no longer part of the expected response to the API.